src/auth/digest/Config.cc

```diff
@@ -774,7 +771,8 @@ Auth::Digest::Config::decode(char const *proxy_auth, const HttpRequest *request,
         }
 
         /* find type */
-        const http_digest_attr_type t = DigestFieldsLookupTable.lookup(keyName);
+        static const auto digestFieldsLookupTable = new LookupTable<http_digest_attr_type>(DIGEST_INVALID_ATTR, DigestAttrs);
```

Comment:
Does not this line still suffer from the same underlying problem, but applied to DigestAttrs global? If it does, then it may be best to create a no-globals style function after all, and move DigestAttrs into that function!

Reply:
I think the DigestAttrs global is safe:
looking at where it lies in the resulting binary (on MacOS)
```
00000001006c05f0 (__DATA_CONST,__const) non-external __ZL11DigestAttrs
```

I found a reference in https://github.com/golang/go/issues/38830 to what it means, and it seems safe enough. The main question is how reliable and portable this is; it might be worthwhile to move everything there for portability reasons

Reply:
> I think the DigestAttrs global is safe

I think it may be safe because LookupTableRecord is currently a constant POD. However, I see no reason not to encapsulate that used-once constant while we are fixing initialization of the global that uses that constant. Your call though.

Reply:
I'll encapsulate it. If anything because I feel it offers a stronger guarantee.